lib/rspec/buildkite/annotation_formatter.rb

```diff
@@ -26,9 +26,16 @@ def initialize(output)
       end
     end
 
+    def example_passed(notification)
+      return if @queue.nil? || RSpec.world.wants_to_quit
+
+      @queue.push(notification) if RSpec.configuration.only_failures?
```

Comment:
We are changing the retry mechanism in Test Splitting `v.09.0`. Previously, we retry the tests by appending `--only-failures` option. With the new changes, we will pass the failed test explicitly to the rspec 
```
rspec failed_spec.rb[1:2] other_failed_spec[2:1:2]
```

Reply:
For context: https://github.com/buildkite/test-splitter/pull/156

Reply:
i love this change though, sorry we moved the splitter implementation out from under you @catkins!

Reply:
Haha, thats all good, I was mostly poking it because i was curious about how it works!

As I mentioned on Slack, it could be cool for the test-splitter to expose an environment variable or something that we can detect in the test collectors and/or this rspec formatter that can behave differently on retry (triggering code paths like this one).

Reply:
You can customize the retry command with test-splitter using `BUILDKITE_SPLITTER_RETRY_CMD`. If we want to expose an environment variable to tell this formatter/test collectors that the run is a retry run, we can do like as below. It's not very clean tho.
```
BUILDKITE_SPLITTER_RETRY_CMD="RSPEC_RETRY=1 bin/rspec {{testExamples}}"
```